wolfssl-sys/build.rs

```diff
@@ -40,6 +40,37 @@ fn copy_wolfssl(dest: &str) -> std::io::Result<()> {
     Ok(())
 }
 
+const PATCHES: &[&str] = &[
+    "patches/0001-DoHelloVerifyRequest-only-do-DTLS-1.3-version-check.patch",
+    "patches/0002-DTLS-1.3-move-state-machine-forward-when-HVR-receive.patch",
+    "patches/0003-DtlsShouldDrop-don-t-ignore-app-data-sent-before-a-S.patch",
+    "patches/0004-Dtls13GetRnMask-Correctly-get-chacha-counter-on-BE-s.patch",
+    "patches/1000-Guard-away-properly-don-t-build-sphincs-code.patch",
+    "patches/1001-Allow-easily-disabling-of-SPHINCS.patch",
+    "patches/2000-Rename-utils.c-to-utils.h.patch",
+    "patches/2001-Merge-pull-request-6700-from-julek-wolfssl-dtls13-do.patch",
+];
+
+/**
+ * Apply patch to wolfssl-src
+ */
+fn apply_patch(dest: &str, patch: &str) {
+    let wolfssl_path = format!("{dest}/wolfssl-src");
```

Comment:
Strictly speaking I think you should emit a [`[cargo:rerun-if-changed=PATH](https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorerun-if-changedpath)`)https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed) for each patch file.

But we aren't retriggering if the wolfssl-src code itself changes (and I think that would be hard to achieve since it is a _lot_ of files) so maybe we can just gloss over this.

This is slightly compounding the kinds of issues I was worried about in https://github.com/expressvpn/wolfssl-sys/pull/53#discussion_r1332631073 (which I carried over into #39 here), one day maybe we'll want to revisit this but for now I think we can cope.

Reply:
Yep, you are right about wolfssl source.

If we think logically, this may not be needed for patches.Whenever we add a new patch, we will update the build script and it will rebuild. That's why we have a static list of patches in script instead of reading the directory and apply all patches.
And we will almost never hand edit a patch. Either add it or remove it.

But anyway I have added for completeness.

Reply:
(One of) the problem(s) is reapplying the same patch a second time will fail. But anyway, the status quo is already fragile IMHO and the changes here don't change that so :shrug: